### PR TITLE
Oauth Support using Keycloak

### PR DESCRIPTION
**Pull request by IOhacker**
## Description
Oauth Support using Keycloak

## Related issues and discussion
#{Issue Number}

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] If you have multiple commits please combine them into one commit by squashing them.

- [ ] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

### CODE REVIEW

File: src/app/core/authentication/authentication.interceptor.ts [hunk 1]
```diff
@@ -18,6 +18,7 @@ const httpOptions = {
 
 /** Authorization header. */
 const authorizationHeader = 'Authorization';
+const authorizationTenantHeader = 'fineract-platform-tenantid';
```
Comment (adamsaghy) [inline src/app/core/authentication/authentication.interceptor.ts:21]: I think it is `Fineract-Platform-TenantId`

File: src/app/core/authentication/authentication.service.ts [hunk 2]
```diff
@@ -94,11 +94,16 @@ export class AuthenticationService {
 
     if (environment.oauth.enabled) {
       let httpParams = new HttpParams();
-      httpParams = httpParams.set('client_id', 'community-app');
+      httpParams = httpParams.set('username', loginContext.username);
+      httpParams = httpParams.set('password', loginContext.password);
+      httpParams = httpParams.set('client_id', `${environment.oauth.appId}`);      
+      httpParams = httpParams.set('client_secret', `${environment.oauth.clientSecret}`);
```
Comment (adamsaghy) [inline src/app/core/authentication/authentication.service.ts:100]: If the grant is password, i dont think we need client_secret:
`client_secret= - (optional) - If the application is a “confidential client” (not a mobile or JavaScript app), then the secret is included as well.`

**Comment by adamsaghy** [comment]
@IOhacker I might not be the right person to review, as my OAuth flow understanding in a frontend application is limited.
However I dont think storing and sending username, password, client id, client secret is okay... i think frontend should use the Authorization Code Flow with PKCE (Proof Key for Code Exchange) which is designed specifically for public clients where the app cannot securely store secrets.

**Review by adamsaghy — CHANGES_REQUESTED** [review]
Kindly see my review...

File: src/app/core/authentication/authentication.interceptor.ts (hunk 1 above)
Comment (IOhacker) [reply src/app/core/authentication/authentication.interceptor.ts]: Fixed

File: src/app/core/authentication/authentication.service.ts (hunk 2 above)
Comment (IOhacker) [reply src/app/core/authentication/authentication.service.ts]: Fixed

**Pull request closed without merging**